notebook/prepare.sh

```diff
@@ -2,16 +2,17 @@
 
 set -x
 
-# We start by adding extra apt packages, since pip modules may required library
-if [ "$EXTRA_APT_PACKAGES" ]; then
-    echo "EXTRA_APT_PACKAGES environment variable found.  Installing."
-    apt update -y
-    apt install -y $EXTRA_APT_PACKAGES
+if [ $UID -eq 0 ]; then
+    # We start by adding extra apt packages, since pip modules may require system dependencies
+    if [ "$EXTRA_APT_PACKAGES" ]; then
+        echo "EXTRA_APT_PACKAGES environment variable found.  Installing."
+        apt update -y
+        apt install -y $EXTRA_APT_PACKAGES
+    fi
+    # Start the script over again, this time as a non-root user.
+    exec sudo -E --user "$NB_USER" PATH="$PATH" "$0" -- "$@"
```

Comment:
Should we also call exit here after starting the script over? Otherwise if JupyterLab is killed the script will continue on and start Jupyter Lab as root.

Reply:
Good point, if that exec statement failed, the script would continue. I'll fix that.